app/presenters/candidate_interface/personal_details_review_presenter.rb

```diff
@@ -148,24 +148,68 @@ def english_language_details_row
       }
     end
 
-    def right_to_work_row
+    def right_to_work_rows
       return nil if british_or_irish?
 
-      {
-        key: I18n.t('application_form.personal_details.right_to_work.label'),
-        value: formatted_right_to_work_or_study,
-        action: (if @editable
-                   {
-                     href: candidate_interface_edit_right_to_work_or_study_path(return_to_params),
-                     visually_hidden_text: I18n.t('application_form.personal_details.right_to_work.change_action'),
-                   }
-                 end),
-        html_attributes: {
-          data: {
-            qa: 'personal_details_right_to_work_or_study',
+      if @application_form.recruitment_cycle_year >= NationalitiesForm::NEW_RIGHT_TO_WORK_FLOW_STARTS
+        @immigration_right_to_work_form = CandidateInterface::ImmigrationRightToWorkForm.build_from_application(
+          @application_form,
+        )
+        @immigration_route_form = CandidateInterface::ImmigrationRouteForm.build_from_application(
```

Comment:
Do we need to instantiate these objects here? Can we not just use the attrs on the application form? Other than `@immigration_right_to_work_form.right_to_work_or_study?`

It seems like they're just being built from the app form and `immigration_right_to_work_form.right_to_work_or_study?` is just casting `immigration_right_to_work` attr in the form.

Can just be  `application_form. immigration_right_to_work`

Wdyt?

Reply:
I think you are right. Was following the established pattern without thinking about it too much.